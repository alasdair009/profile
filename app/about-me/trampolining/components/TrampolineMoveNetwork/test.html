<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
