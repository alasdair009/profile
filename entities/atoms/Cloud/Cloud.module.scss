@use "../../../app/styles/dimensions";
@use "../../../app/styles/radii";
@use "../../../app/styles/helpers/rem";
@use "sass:map";

.root {
  background: var(--skyColor);
  overflow: hidden;
  position: relative;
}

.cloudCircle {
  border-radius: map.get(radii.$borderRadii, round);
  box-shadow: 0 #{rem.toRem(map.get(dimensions.$sizes, s256) * 1.5)}
    var(--dispersion) 0 var(--cloudColor);
  filter: url(#cloudFilter);
  height: rem.toRem(map.get(dimensions.$sizes, s256));
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -200%) translateZ(0);
  width: rem.toRem(map.get(dimensions.$sizes, s512));
}
