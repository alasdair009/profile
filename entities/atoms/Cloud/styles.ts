"use client";

import styled from "styled-components";
import { colors, sizes } from "@/entities";
import { rem } from "polished";

export const Root = styled.div`
  background: linear-gradient(165deg, #527785 0%, #7fb4c7 100%);
  overflow: hidden;
  position: relative;
`;

export const CloudCircle = styled.div<{ $dispersion: number }>`
  border-radius: 50%;
  box-shadow: 0 ${rem(sizes.s256.raw * 1.5)}
    ${({ $dispersion }) => rem($dispersion)} 0 ${colors.whiteGhost};
  filter: url(#cloudFilter);
  height: ${sizes.s256.rem};
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -200%) translateZ(0);
  width: ${sizes.s512.rem};
`;
