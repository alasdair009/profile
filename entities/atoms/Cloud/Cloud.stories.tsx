import { Meta, StoryObj } from "@storybook/react";
import { Cloud } from "./Cloud";
import { expect, within } from "@storybook/test";

const meta: Meta<typeof Cloud> = {
  component: Cloud,
  args: {
    scale: 180,
    style: {
      height: 500,
    },
  },
  argTypes: {
    dispersion: {
      control: { type: "range", min: 10, max: 150, step: 1 },
    },
    scale: {
      control: { type: "range", min: 180, max: 500, step: 1 },
    },
  },
};
export default meta;

export const Default: StoryObj<typeof Cloud> = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const cloudElement = canvas.getByTestId(Cloud.name);

    await expect(cloudElement).toBeInTheDocument();
  },
};
