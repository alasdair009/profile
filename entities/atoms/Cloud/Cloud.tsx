import { CloudCircle, Root } from "./styles";
import { HTMLAttributes } from "react";

type CloudProps = {
  scale?: number;
  dispersion: number;
} & HTMLAttributes<HTMLDivElement>;

/**
 * Sky and cloud generated using CSS.
 */
export function Cloud({ dispersion = 60, scale = 180, ...rest }: CloudProps) {
  return (
    <Root data-testid={Cloud.name} {...rest}>
      <CloudCircle $dispersion={dispersion} />
      <svg width="0" height="0">
        <filter id="cloudFilter">
          <feTurbulence
            type="fractalNoise"
            baseFrequency={0.01}
            numOctaves={10}
          />
          <feDisplacementMap in="SourceGraphic" scale={scale} />
        </filter>
      </svg>
    </Root>
  );
}
