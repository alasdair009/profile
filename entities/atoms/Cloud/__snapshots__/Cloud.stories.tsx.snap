// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Cloud Default play-test 1`] = `
<div data-testid="Cloud"
     class="sc-drnuxz hHgRlH"
     style="height: 500px;"
>
  <div class="sc-hDcvty foLBEi">
  </div>
  <svg width="0"
       height="0"
  >
    <filter id="cloudFilter">
      <feTurbulence type="fractalNoise"
                    basefrequency="0.01"
                    numoctaves="10"
      >
      </feTurbulence>
      <feDisplacementMap in="SourceGraphic"
                         scale="180"
      >
      </feDisplacementMap>
    </filter>
  </svg>
</div>
`;
